Applications/Custom/mae_loss.cpp

```diff
@@ -12,8 +12,54 @@
  */
 #include "mae_loss.h"
 
+#include <cmath>
+
+#include <tensor.h>
 namespace custom {
-const std::string MaeLossLayer::type = "mae_loss";
+
+int MaeLossLayer::initialize(nntrainer::Manager &manager) {
+  output_dim = input_dim;
+  return ML_ERROR_NONE;
+}
+
+int MaeLossLayer::setProperty(std::vector<std::string> values) {
+  /// this implementation makes to pass the test, this will change soon.
+  return values.size();
+}
+
+void MaeLossLayer::forwarding(bool training) {
+  nntrainer::Tensor &label = net_hidden[0]->getGradientRef();
+  nntrainer::Tensor &predicted = net_input[0]->getVariableRef();
+  nntrainer::Tensor &output = net_hidden[0]->getVariableRef();
+
+  bool with_label = !label.uninitialized();
+  if (with_label) {
+    predicted.subtract(label, output);
+    output.apply_i([](float x) { return x > 0 ? x : -x; });
+  } else {
+    output.fill(predicted);
+  }
+}
+
+void MaeLossLayer::calcDerivative() {
+  nntrainer::Tensor &predicted = net_input[0]->getVariableRef();
+  nntrainer::Tensor &label = net_hidden[0]->getGradientRef();
+
+  nntrainer::Tensor &deriv = net_input[0]->getGradientRef();
+
+  /// This can be saved at MaeLossLayer::forwarding, but this is done here on
+  /// purpose for demonstration purpose
+  predicted.subtract(label, deriv);
+
+  deriv.apply_i([](float x) {
+    if (fabs(x) < 1e-7) {
```

Comment:
please declare a layer const `epsilon` and use here?

Reply:
Updated.